javaparser-core/src/main/javacc/java_1_8.jj

```diff
@@ -153,6 +153,23 @@ final class ASTParser {
 	    return ret;
     }
 
+    private ReferenceType juggleArrayTypes(Range range, Type type, int arrayCount, List<List<AnnotationExpr>> arrayAnnotations) {
+        if (arrayCount == 0) {
+            return new ReferenceType(range, type);
```

Comment:
`type` can only be a ClassOrInterfaceType. You should cast it to `ReferenceType` rather than wrapping it.